config/reader.go

```diff
@@ -42,6 +44,40 @@ const (
 	LocalPackagesSuffix = "-local"
 )
 
+func parseOsRelease(body *bufio.Reader) map[string]string {
+	ret := make(map[string]string)
+	for {
+		line, err := body.ReadString('\n')
+		if err != nil {
+			return ret
+		}
+		key, value, _ := strings.Cut(line, "=")
+		// note: we trim `value` rather than `line` because os_version value is quoted sometimes.
+		ret[key] = strings.Trim(value, "\n\"")
+	}
+}
+
+// This reads the granular platform constraints (os version, distro, etc).
+// This further constrains the basic runtime.GOOS/GOARCH stuff in getAgentInfo
+// so module authors can publish builds with ABI or SDK dependencies. The
+// list of tags returned by this function is expected to grow.
+func readExtendedPlatformTags() []string {
+	// TODO(APP-6696): CI in multiple environments (alpine + mac), darwin support.
+	tags := make([]string, 0, 4)
+	if runtime.GOOS == "linux" && rutils.PathExists("/etc/os-release") {
+		if body, err := os.Open("/etc/os-release"); err != nil {
+			logging.Global().Errorw("can't open /etc/os-release, modules may not load correctly", "err", err)
+		} else {
+			defer body.Close() //nolint:errcheck
+			osRelease := parseOsRelease(bufio.NewReader(body))
+			tags = append(tags, "distro:"+osRelease["ID"])
+			tags = append(tags, "os_version:"+osRelease["VERSION_ID"])
+			tags = append(tags, "codename:"+osRelease["VERSION_CODENAME"])
```

Comment:
No guarantee that all of these will be in every os-release file, as the fields are all optional. Many distros populate only some of these. If they're blank, you're gonna end up with odd tags like `codename:` which may break other stuff down the line when parsing.

Ex: Arch looks like this:

```
NAME="Arch Linux"
PRETTY_NAME="Arch Linux"
ID=arch
BUILD_ID=rolling
ANSI_COLOR="38;2;23;147;209"
HOME_URL="https://archlinux.org/"
DOCUMENTATION_URL="https://wiki.archlinux.org/"
SUPPORT_URL="https://bbs.archlinux.org/"
BUG_REPORT_URL="https://gitlab.archlinux.org/groups/archlinux/-/issues"
PRIVACY_POLICY_URL="https://terms.archlinux.org/docs/privacy-policy/"
LOGO=archlinux-logo
```

Reply:
changed logic in a532b41 to only set a tag if the value exists